webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -182,23 +223,220 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
+                  key={logEntry.id}
+                  level={logEntry.level}
+                  message={logEntry.message}
+                  style={{
+                    backgroundColor:
+                      logEntry.id === selectedLog?.id ? 'var(--theme-ix-active)' : 'transparent',
+                  }}
+                />
+              </div>
+            ))
+          ) : (
+            <Message icon="warning" title="No logs to show. " />
+          )}
+        </div>
+      </div>
+    );
+  }, [settings.enableRegex, formattedSearchResults, container, selectedLog, updateSettings]);
+
+  const formatClipboardHeader = (log: Log): string => {
+    const logEntry = formatLogEntry(log);
+    const format = `%${MAX_DATETIME_LENGTH - 1}s `;
+    const level = `<${logEntry.level || ''}>`;
+    return sprintf(`%-9s ${format}`, level, logEntry.formattedTime);
+  };
+
+  const clipboardCopiedMessage = useMemo(() => {
+    const linesLabel = pluralizer(logs.length, 'entry', 'entries');
+    return `Copied ${logs.length} ${linesLabel}!`;
+  }, [logs]);
+
+  const getClipboardContent = useCallback(() => {
+    return logs.map((log) => `${formatClipboardHeader(log)}${log.message || ''}`).join('\n');
+  }, [logs]);
+
+  const handleFullScreen = useCallback(() => {
+    if (logsRef.current && screenfull.isEnabled) screenfull.toggle();
+  }, []);
+
+  const rightButtons = (
+    <Row>
+      <ClipboardButton copiedMessage={clipboardCopiedMessage} getContent={getClipboardContent} />
+      <Button
+        aria-label="Toggle Fullscreen Mode"
+        icon={<Icon name="fullscreen" showTooltip title="Toggle Fullscreen Mode" />}
+        onClick={handleFullScreen}
+      />
+      <Button
+        aria-label="Download Logs"
+        icon={<Icon name="download" showTooltip title="Download Logs" />}
+        onClick={handleDownloadLogs}
+      />
+    </Row>
+  );
+
   return (
-    <div className={css.base}>
+    <div className={css.base} ref={container}>
       <Spinner conditionalRender spinning={!trial}>
-        <LogViewer
-          decoder={mapV1LogsResponse}
-          serverAddress={serverAddress}
-          title={logFilters}
-          onDownload={handleDownloadLogs}
-          onError={handleError}
-          onFetch={handleFetch}
+        <div className={css.header}>
+          <div className={css.filters}>
+            <Input
+              allowClear
+              placeholder="Search Logs..."
+              value={searchInput || settings.searchText}
+              width={240}
+              onChange={onSearchChange}
+            />
+            <Tooltip content={`${searchOn ? 'Close' : 'Open'} Search`}>
+              <Button
+                type={searchOn ? 'primary' : 'default'}
+                onClick={() => {
+                  setSearchOn((prev) => !prev);
+                  searchOn && setLogViewerOn(true);
+                }}>
+                <Icon decorative name="search" />
+              </Button>
+            </Tooltip>
```

Comment:
even though there's a tooltip here, we still need to label the button. As such, the icons aren't decorative. It might make sense to instead use the showTootlip and title props:
```suggestion
            <Button
              type={searchOn ? 'primary' : 'default'}
              onClick={() => {
                setSearchOn((prev) => !prev);
                searchOn && setLogViewerOn(true);
              }}>
              <Icon showTooltip title={`${searchOn ? 'Close' : 'Open'} Search`} name="search" />
            </Button>
```